suite-common/wallet-transactions/src/transactionReducer.ts

```diff
@@ -0,0 +1,140 @@
+            const accountTxs = initializeAccount(state, key);
+            const index = accountTxs.findIndex(t => t && t.txid === txid);
+            if (accountTxs[index]) accountTxs[index] = tx;
+        })
+        .addCase(transactionActions.removeTransaction, (state, { payload }) => {
+            const { account, txs } = payload;
+            const transactions = state.transactions[account.key] || [];
+            txs.forEach(tx => {
+                const index = transactions.findIndex(t => t.txid === tx.txid);
+                transactions.splice(index, 1);
+            });
+        })
+        .addCase(transactionActions.addTransaction, (state, { payload }) => {
+            const { transactions, account, page } = payload;
+            if (transactions.length < 1) return;
+            initializeAccount(state, account.key);
+            const accountTxs = state.transactions[account.key];
+
+            if (!accountTxs) return;
+
+            transactions.forEach((transaction, i) => {
+                // first we need to make sure that transaction is not undefined, then check if transactionid matches
+                const existingTx = findTransaction(transaction.txid, accountTxs);
+                if (!existingTx) {
+                    // add a new transaction
+                    if (page) {
+                        // insert a tx object at correct index
+                        const txIndex = (page - 1) * settingsCommonConfig.TXS_PER_PAGE + i;
+                        accountTxs[txIndex] = transaction;
+                    } else {
+                        // no page arg, insert the tx at the beginning of the array
+                        accountTxs.unshift(transaction);
+                    }
+                } else {
+                    // update the transaction if conditions are met
+                    const existingTxIndex = accountTxs.findIndex(
+                        t => t && t.txid === existingTx.txid,
+                    );
+
+                    if (
+                        (!existingTx.blockHeight && transaction.blockHeight) ||
+                        (!existingTx.blockTime && transaction.blockTime)
+                    ) {
+                        // pending tx got confirmed (blockHeight changed from undefined/0 to a number > 0)
+                        accountTxs[existingTxIndex] = { ...transaction };
+                    }
+                }
+            });
+        })
+        .addMatcher(
+            matchesActionType(extra.actionTypes.storageLoad),
+            extra.reducers.storageLoadTransactions,
+        )
+        .addMatcher(
+            matchesActionType(extra.actionTypes.fiatRateUpdate),
+            extra.reducers.txFiatRateUpdate,
+        )
+        .addMatcher(
+            matchesActionType(extra.actionTypes.removeAccount),
+            extra.reducers.removeAccountsTransactions,
+        );
+});
+
+export const selectIsLoading = (state: TransactionSliceRootState) =>
```

Comment:
use better names because these selector are exported out of this file and in context of wallet isLoading can mean anything

`selectIsLoadingTransactions`
`selectTransactionsError`

etc.

Reply:
Yeah good point. I was thinking about it and only kept it like this because you always assign the selector value to another variable but it could be confusing when also importing other `isLoading` variable to same file. Renamed.

Reply:
https://github.com/trezor/trezor-suite/pull/5981/commits/86973889f127ee510e9e5b3f9fd0d10ad515d1c8